src/CloudGraphsDFG/services/CloudGraphsDFG.jl

```diff
@@ -51,31 +12,96 @@ function _getDuplicatedEmptyDFG(dfg::CloudGraphsDFG)::CloudGraphsDFG
+        dfg.encodePackedTypeFunc,
+        dfg.getPackedTypeFunc,
+        dfg.decodePackedTypeFunc,
+        dfg.rebuildFactorMetadata!,
+        solverParams=deepcopy(dfg.solverParams))
 end
 
 # Accessors
-getLabelDict(dfg::CloudGraphsDFG) = dfg.labelDict
-getDescription(dfg::CloudGraphsDFG) = dfg.description
-setDescription!(dfg::CloudGraphsDFG, description::String) = dfg.description = description
-getAddHistory(dfg::CloudGraphsDFG) = dfg.addHistory
-getSolverParams(dfg::CloudGraphsDFG) = dfg.solverParams
-function setSolverParams!(dfg::CloudGraphsDFG, solverParams::T)::T where T <: AbstractParams
-    return dfg.solverParams = solverParams
+function getDescription(dfg::CloudGraphsDFG)
+    return _getNodeProperty(
+        dfg.neo4jInstance,
+        [dfg.sessionId, dfg.robotId, dfg.userId, "SESSION"],
+        "description")
+end
+function setDescription!(dfg::CloudGraphsDFG, description::String)
+    _setNodeProperty(
+        dfg.neo4jInstance,
+        [dfg.sessionId, dfg.robotId, dfg.userId, "SESSION"],
+        "description",
+        description)
 end
 
 function getSerializationModule(dfg::CloudGraphsDFG)::Module where G <: AbstractDFG
     # TODO: If we need to specialize this for RoME etc, here is where we'll change it.
     return Main
 end
 
-function exists(dfg::CloudGraphsDFG, nId::Symbol)
+##==============================================================================
+## User/Robot/Session Data CRUD
+##==============================================================================
+
+
+function getUserData(dfg::CloudGraphsDFG)::Dict{Symbol, String}
+    propVal = _getNodeProperty(dfg.neo4jInstance, _getLabelsForType(dfg, User), "data")
+    return JSON2.read(String(base64decode(propVal)), Dict{Symbol, String})
+end
+function setUserData!(dfg::CloudGraphsDFG, data::Dict{Symbol, String})::Bool
+    count = _setNodeProperty(dfg.neo4jInstance, [dfg.userId, "USER"], "data", base64encode(JSON2.write(data)))
+    return count == 1
+end
+function getRobotData(dfg::CloudGraphsDFG)::Dict{Symbol, String}
+    propVal = _getNodeProperty(dfg.neo4jInstance, [dfg.userId, dfg.robotId, "ROBOT"], "data")
+    return JSON2.read(String(base64decode(propVal)), Dict{Symbol, String})
+end
+function setRobotData!(dfg::CloudGraphsDFG, data::Dict{Symbol, String})::Bool
+    count = _setNodeProperty(dfg.neo4jInstance, [dfg.userId, dfg.robotId, "ROBOT"], "data", base64encode(JSON2.write(data)))
+    return count == 1
+end
+function getSessionData(dfg::CloudGraphsDFG)::Dict{Symbol, String}
+    propVal = _getNodeProperty(dfg.neo4jInstance, [dfg.userId, dfg.robotId, dfg.sessionId, "SESSION"], "data")
+    return JSON2.read(String(base64decode(propVal)), Dict{Symbol, String})
+end
+function setSessionData!(dfg::CloudGraphsDFG, data::Dict{Symbol, String})::Bool
+    count = _setNodeProperty(dfg.neo4jInstance, [dfg.userId, dfg.robotId, dfg.sessionId, "SESSION"], "data", base64encode(JSON2.write(data)))
+    return count == 1
+end
+
+# New API
```

Comment:
I couldn't make my mind up if we have to support this kind of operations on user data.
I think we can perhaps just for example `getUserData`, change `setUserData!`?

Reply:
Personally I feel user data is a chunk of whatever a user wants, no crud is needed. If that works, then yep we can remove the new API?

Reply:
I’m ok with just get and set on cloud.